src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal.hidden !== true && !(animal instanceof Carnivore)) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        const id = Animal.alive.indexOf(animal);
+        
+        Animal.alive.splice(id, 1);
+      }
```

Comment:
Don't manipulate with array in this way, it's better to use filter method instead